### PR TITLE
OV-2: FEAT: Sign-in flow

### PR DESCRIPTION
**Pull request by JKaypa**
Add sign-in functionality from the backend. A data transfer object containing user credentials (email, password) is used to compare against existing data

### CODE REVIEW

File: backend/src/bundles/users/user.repository.ts [hunk 1]
```diff
@@ -9,8 +9,13 @@ class UserRepository implements Repository {
         this.userModel = userModel;
     }
 
-    public find(): ReturnType<Repository['find']> {
-        return Promise.resolve(null);
+    public async find(payload: string | number): Promise<UserEntity | null> {
```
Comment (anton-otroshchenko) [inline backend/src/bundles/users/user.repository.ts:12]: find method should only find by id.

Please create a specific method if you want to use different fields

for example findByEmail()

**Review by anton-otroshchenko — CHANGES_REQUESTED** [review]

File: backend/src/common/types/repository.type.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 type Repository<T = unknown> = {
     find(): Promise<T>;
     findAll(): Promise<T[]>;
+    findByEmail(email: string): Promise<T>;
```
Comment (anton-otroshchenko) [inline backend/src/common/types/repository.type.ts:4]: we will have this method only in user repo so there is no need to put this in type here

File: backend/src/common/types/service.type.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@ type Service<T = unknown> = {
     findAll(): Promise<{
         items: T[];
     }>;
+    findByEmail(email: string): Promise<T>;
```
Comment (anton-otroshchenko) [inline backend/src/common/types/service.type.ts:6]: same here

**Review by anton-otroshchenko — CHANGES_REQUESTED** [review]

**Review by anton-otroshchenko — APPROVED** [review]

File: backend/src/bundles/auth/auth.controller.ts [hunk 4]
```diff
@@ -65,6 +83,17 @@ class AuthController extends BaseController {
      *                    type: object
      *                    $ref: '#/components/schemas/User'
      */
+    private async signIn(
```
Comment (nikita-remeslov) [inline backend/src/bundles/auth/auth.controller.ts:86]: you should add js doc for endpoint

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**